### PR TITLE
prplmesh-be: Implement set function for AccessPoint

### PR DESCRIPTION
**Pull request by VladyslavTupikin**
Need to implement new function for setting and applying data for Controller.Network.AccessPoint object.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

### CODE REVIEW

File: src/prplmesh_set.lua [hunk 1]
```diff
@@ -68,6 +68,151 @@
...
+    local ap_param_tbl = {}
+    local param_idx = 2
+
+    local empty_param = false
+    while param_idx <= #args do
+        empty_param = false
+        if args[param_idx] == "-pname" and args[param_idx+2] == "-pvalue" then
+
+            if args[param_idx+1] and args[param_idx+3] ~= "-pname" then
+                ap_param_tbl[ args[param_idx + 1] ] = args[param_idx+3]
```
Comment (MuteSpirit) [inline src/prplmesh_set.lua]: Please use spaces before and after plus (+) sign: ```param_idx + 3```

File: src/prplmesh_set.lua [hunk 2]
```diff
@@ -68,6 +68,151 @@
...
+            else
+                empty_param = true
+                ap_param_tbl[ args[param_idx + 1] ] = ""
+            end
+
+            if args[param_idx+3] == nil then
```
Comment (MuteSpirit) [inline src/prplmesh_set.lua]: Please use spaces before and after plus (+) sign: ```param_idx + 3```

File: src/prplmesh_set.lua [hunk 3]
```diff
@@ -68,6 +68,151 @@
...
+    @brief  Function for converting CMD arguments to key-value table.
+
+    @param  args Command Line arguments.
+            Example: Controller.Network.AccessPoint 1 -pname SSID -pvalue WiFi
+
+    @return Table which contains list of parameters and values otherwise false.
+--]]
+local function parse_args(args)
+    local ap_param_tbl = {}
+    local param_idx = 2
```
Comment (MuteSpirit) [inline src/prplmesh_set.lua:86]: It is _hidden function knowledge about call context_ that it's needed to skip first argument
By my opinion you should pass into function
* index of first parsing argument: ```parse_args(2)```
* table of arguments to parse:
```Lua
table.remove(args, 1)
parse_args(args)
```
(but i'm not about 2nd variant correctness in view of Lua best practice view)

File: src/prplmesh_set.lua [hunk 4]
```diff
@@ -68,6 +68,151 @@
...
+local function set_data(path, ap_params)
+
+    if not path or not ap_params then
+        error("Bad path or ap_params.")
+        return false
+    end
+
+    local data = {parameters = ap_params}
+
+    local _ubus_connection = ubus.connect()
```
Comment (MuteSpirit) [inline src/prplmesh_set.lua]: Why is it needed to open connection to U-Bus explicitly if `call_ubus` will do that inside itself?

File: src/prplmesh_set.lua [hunk 5]
```diff
@@ -68,6 +68,151 @@
...
+
+    local ret = tostring(ing.ResCode.FAIL) .. ";" .. tostring(ing.StatCode.OK) .. ";"
+
+    if #args < 2 then
+        error("Wrong arguments: " .. tostring(#args))
+        print(ret)
+        return ret
+    end
+
+    local ap_path = args[1]
```
Comment (MuteSpirit) [inline src/prplmesh_set.lua:174]: Just suggestion: if you apply only AccessPoint then maybe filter another objects and stop here if object is not AccessPoint?

**Review by MuteSpirit — CHANGES_REQUESTED** [review]

File: src/prplmesh_set.lua (hunk 1 above)
Comment (VladyslavTupikin) [reply src/prplmesh_set.lua]: fixed

File: src/prplmesh_set.lua (hunk 3 above)
Comment (VladyslavTupikin) [reply src/prplmesh_set.lua:86]: Current solution works well.

File: src/prplmesh_set.lua (hunk 2 above)
Comment (VladyslavTupikin) [reply src/prplmesh_set.lua]: fixed

File: src/prplmesh_set.lua (hunk 5 above)
Comment (VladyslavTupikin) [reply src/prplmesh_set.lua:174]: Need to rename script in feature and modify DB.

File: src/prplmesh_set.lua (hunk 4 above)
Comment (VladyslavTupikin) [reply src/prplmesh_set.lua]: Don't need. removed.

**Review by MuteSpirit — APPROVED** [review]

**Pull request merged**